### ISSUE TITLE
Optimize Querying of RequestCache -> continue parameter

### ISSUE THREAD

**Issue by Writtscher** [spring-projects/spring-security#13438]
**Expected Behavior**

Do not append endless continue parameters.

**Current Behavior**

Adds endless continue parameters

**Context**

I have implemented a simple oidc login to play with the new api / version. Whenever the login succeeds I see a continue parameter. I understand why this is the case: https://docs.spring.io/spring-security/reference/5.8/migration/servlet/session-management.html#requestcache-query-optimization . But I do not understand why it does append everytime I remove the jsessionid another continue parameter. I mean if the url already contains the continue parameter than there is no reason to append another one. Or am I mistaken?

In short: When I login I am redirected to my requested url http://localhost:8080/xxx?continue. When I remove the jsessionid and reload the page I am redirected to http://localhost:8080/xxx?continue&continue. When I repeat it I am redirected to http://localhost:8080/xxx?continue&continue&continue.

**Comment by jzheaux** [comment]
Thanks for the report, @Writtscher. I've pushed a fix, and it should go out with the next set of releases.
